compiler/ccgtypes.nim

```diff
@@ -800,11 +800,6 @@ proc getRecordDesc(m: BModule; typ: PType, name: Rope,
     if not hasField and typ.itemId notin m.g.graph.memberProcsPerType:
       if desc == "":
         result.add("\tchar dummy;\n")
-      elif typ.len == 1 and typ.n[0].kind == nkSym:
-        let field = typ.n[0].sym
-        let fieldType = field.typ.skipTypes(abstractInst)
-        if fieldType.kind == tyUncheckedArray:
-          result.add("\tchar dummy;\n")
```

Comment:
The C spec requires this field IIRC.

Reply:
I understand but an array of length 1 takes up more memory than necessary and it's usually more than 1 byte.

Reply:
I observe the generated c code. when allocating memory for `UncheckedArray`, it does not allocate an extra byte, it just allocates according to the size required by the user. Before my modification, the dummy field occupies a word that should belong to the data field section, causing the data field to be smaller than the size the user wants. sorry for my slow reply because my english is not very good

```c
N_LIB_PRIVATE N_NIMCALL(tyObject_ChunkObj__mOTFrUaGXmv4xFi10w76AA*, alloc__spaces_u3)(NI size_p0) {
	tyObject_ChunkObj__mOTFrUaGXmv4xFi10w76AA* result;
	nimfr_("alloc", "/root/repos/spaces/src/spaces.nim");
{	result = NIM_NIL;
	nimln_(11);	eqdestroy___spaces_u32(result);
	if ((size_p0) < ((NI)0) || (size_p0) > ((NI)IL64(9223372036854775807))){ raiseRangeErrorI(size_p0, ((NI)0), ((NI)IL64(9223372036854775807))); goto BeforeRet_;
	}
	result = (tyObject_ChunkObj__mOTFrUaGXmv4xFi10w76AA*) nimNewObj(((NI) (size_p0)), NIM_ALIGNOF(tyObject_ChunkObj__mOTFrUaGXmv4xFi10w76AA));
	}BeforeRet_: ;
	popFrame();
	return result;
}
```